operator/src/main/java/oracle/kubernetes/operator/steps/InitializeWebhookIdentityStep.java

```diff
@@ -0,0 +1,187 @@
+import static oracle.kubernetes.operator.logging.MessageKeys.WEBHOOK_IDENTITY_INITIALIZATION_FAILED;
+import static oracle.kubernetes.operator.utils.SelfSignedCertUtils.createKeyPair;
+import static oracle.kubernetes.operator.utils.SelfSignedCertUtils.generateCertificate;
+import static oracle.kubernetes.operator.utils.WebhookCertificates.WEBHOOK_CERTIFICATE;
+import static oracle.kubernetes.operator.utils.WebhookCertificates.WEBHOOK_CERTIFICATE_KEY;
+import static oracle.kubernetes.operator.utils.WebhookCertificates.WEBHOOK_DIR;
+
+public class InitializeWebhookIdentityStep extends Step {
+
+  private static final LoggingFacade LOGGER = LoggingFactory.getLogger("Webhook", "Operator");
+  private static final String WEBHOOK_CM = "weblogic-webhook-cm";
+  private static final String WEBHOOK_SECRETS = "weblogic-webhook-secrets";
+  private static final String SHA_256_WITH_RSA = "SHA256withRSA";
+  private static final String COMMON_NAME = "weblogic-webhook";
+  private static final int CERTIFICATE_VALIDITY_DAYS = 3650;
+  private static final File certFile = new File(WEBHOOK_DIR + "/config/webhookCert");
+  private static final File keyFile = new File(WEBHOOK_DIR + "/secrets/webhookKey");
+  public static final String WEBHOOK_KEY = "webhookKey";
+
+  /**
+   * Constructor for the InitializeWebhookIdentityStep.
+   * @param next Next step to be executed.
+   */
+  public InitializeWebhookIdentityStep(Step next) {
+    super(next);
+  }
+
+  @Override
+  public NextAction apply(Packet packet) {
+    try {
+      if (certFile.exists() && keyFile.exists()) {
+        // The webhook's ssl identity has already been created.
+        reuseIdentity();
+        return doNext(getNext(), packet);
+      } else {
+        // The webhook's ssl identity hasn't been created yet.
+        return createIdentity(packet);
+      }
+    } catch (Exception e) {
+      LOGGER.warning(WEBHOOK_IDENTITY_INITIALIZATION_FAILED, e.toString());
+      throw new RuntimeException(e);
+    }
+  }
+
+  private void reuseIdentity() throws IOException {
+    // copy the certificate and key from the webhook's config map and secret
+    // to the locations the webhook runtime expects
+    FileUtils.copyFile(certFile, new File(WEBHOOK_CERTIFICATE));
+    FileUtils.copyFile(keyFile, new File(WEBHOOK_CERTIFICATE_KEY));
+  }
+
+  private NextAction createIdentity(Packet packet) throws Exception {
+    KeyPair keyPair = createKeyPair();
+    String key = convertToPEM(keyPair.getPrivate());
+    writeToFile(key, new File(WEBHOOK_CERTIFICATE_KEY));
+    X509Certificate cert = generateCertificate(WEBHOOK_CERTIFICATE, keyPair, SHA_256_WITH_RSA, COMMON_NAME,
+            CERTIFICATE_VALIDITY_DAYS);
+    writeToFile(getBase64Encoded(cert), new File(WEBHOOK_CERTIFICATE));
+    // put the new certificate in the webhook's config map so that it will be available
+    // the next time the webhook is started
+    return doNext(recordWebhookCert(WEBHOOK_CERTIFICATE, cert,
```

Comment:
This is the other potential concurrency point. The webhook is about to record its certificate and key, but the create/replace of the Secret may fail with a conflict, in which case the webhook should read the existing data and reuse the identity set by the other webhook instance that was faster.

Reply:
Fixed in 6b88d5b.